src/Microsoft.AspNetCore.Mvc.Formatters.Xml/Internal/MvcXmlDataContractSerializerMvcOptionsSetup.cs

```diff
@@ -21,7 +39,9 @@ public void Configure(MvcOptions options)
         {
             options.ModelMetadataDetailsProviders.Add(new DataMemberRequiredBindingMetadataProvider());
 
-            options.OutputFormatters.Add(new XmlDataContractSerializerOutputFormatter());
+            var outputFormatterLogger = _loggerFactory.CreateLogger<XmlDataContractSerializerOutputFormatter>();
+
+            options.OutputFormatters.Add(new XmlDataContractSerializerOutputFormatter(outputFormatterLogger));
```

Comment:
This is strange. Where else do we create a type-specific logger for another type? I'd expect the `XmlDataContractSerializerOutputFormatter` constructor to get the `ILoggerFactory`.